packages/contracts-bedrock/src/libraries/Encoding.sol

```diff
@@ -173,4 +173,50 @@ library Encoding {
             batcherHash
         );
     }
+
+    /// @notice Returns an appropriately encoded call to L1Block.setL1BlockValuesInterop
+    /// @param baseFeeScalar       L1 base fee Scalar
+    /// @param blobBaseFeeScalar   L1 blob base fee Scalar
+    /// @param sequenceNumber      Number of L2 blocks since epoch start.
+    /// @param timestamp           L1 timestamp.
+    /// @param number              L1 blocknumber.
+    /// @param baseFee             L1 base fee.
+    /// @param blobBaseFee         L1 blob base fee.
+    /// @param hash                L1 blockhash.
+    /// @param batcherHash         Versioned hash to authenticate batcher by.
+    /// @param interopSetSize      Size of the interop dependency set.
+    /// @param chainIds            Array of chain IDs in the interop dependency set.
+    function encodeSetL1BlockValuesInterop(
+        uint32 baseFeeScalar,
+        uint32 blobBaseFeeScalar,
+        uint64 sequenceNumber,
+        uint64 timestamp,
+        uint64 number,
+        uint256 baseFee,
+        uint256 blobBaseFee,
+        bytes32 hash,
+        bytes32 batcherHash,
+        uint8 interopSetSize,
+        uint256[] memory chainIds
```

Comment:
nit: dependency set and also can remove the explicit size and use `dependencySet.length`

Reply:
Should double check the length doesn't overflow uint8